code/modules/mob/living/say.dm

```diff
@@ -204,19 +189,22 @@ proc/get_radio_key_from_channel(var/channel)
 		var/msg
 		if(!speaking || !(speaking.flags & NO_TALK_MSG))
 			msg = "<span class='notice'>\The [src] talks into \the [used_radios[1]]</span>"
-		for(var/mob/living/M in hearers(5, src))
-			if((M != src) && msg)
+
+		if(msg)
+			for(var/mob/living/M in hearers(5, src) - src)
 				M.show_message(msg)
-			if(speech_sound)
-				sound_vol *= 0.5
+
+		if(speech_sound)
+			sound_vol *= 0.5
+
 
 	var/turf/T = get_turf(src)
 
 	//handle nonverbal and sign languages here
 	if(speaking)
 		if(speaking.flags & NONVERBAL)
 			if(prob(30))
-				src.custom_emote(1, "[pick(speaking.signlang_verb)].")
+				custom_emote(1, "[pick(speaking.signlang_verb)].")

```

Comment:
While you're at it: Sign languages (and presumably nonverbal languages) don't work. Replacing `pick(speaking.signlang_verb)` with `pick(verb)` or just `verb` will fix it.